app/controllers/trains_controller.rb

```diff
@@ -3,10 +3,24 @@ def index
     @trains = TrainApi.new.index
   end
 
+  def edit
+    flash["errors"] = nil
+    @train = TrainApi.new.show(params)
+  end
+
   def create
   end
 
-  def edit
+  def update
+    response = TrainApi.new.update(params[:id], params)
+
+    if response.success?
+      redirect_to trains_path
```

Comment:
any success message? we should probably show some kind of message to indicate what happened.

Reply:
i thought the success messages are already in the train_controller in the api?

Reply:
> any success message? we should probably show some kind of message to indicate what happened.

i thought the success messages are already in the train_controller in the api?

Reply:
your not rendering anything to the user.
api gives a message but truck app gives no message.

this was an easy one to spot as when you submit successfully your app shows nothing

Reply:
> api gives a message but truck app gives no message.

so should the message should be visible on the view as well?

Reply:
you need A message to be on the view as its important for UX.
whether or not its the API message is up to you.

but there has to be a message